src/Symfony/Component/OptionsResolver/Debug/OptionResolverIntrospector.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\OptionsResolver\Debug;
+
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ *
+ * @final
+ */
+class OptionResolverIntrospector
+{
+    private $optionsResolver;
+    private $get;
+    private $has;
+
+    public function __construct(OptionsResolver $optionsResolver)
+    {
+        $this->optionsResolver = $optionsResolver;
+
+        $this->get = \Closure::bind(function ($property, $option) {
+            return $this->{$property}[$option];
+        }, $optionsResolver, $optionsResolver);
+
+        $this->has = \Closure::bind(function ($property, $option) {
+            return array_key_exists($option, $this->{$property});
+        }, $optionsResolver, $optionsResolver);
+    }
+
+    /**
+     * @param string $option
+     *
+     * @return mixed|null
+     */
+    public function getDefault($option)
+    {
+        return $this->get('defaults', $option);
+    }
+
+    /**
+     * @param string $option
+     *
+     * @return bool True if a default value was set
+     */
+    public function hasDefault($option)
+    {
+        return $this->has('defaults', $option);
+    }
+
+    /**
+     * @param string $option
+     *
+     * @return callable[]|null Null if none were set
```

Comment:
The `null` part could be replace by throwing an appropriate exception, and would eliminate the need for the `hasDefault()` method. Would it be better? `NoValueSetException`?

Reply:
I think so `NoSuchOptionException`? thrown when trying to read an option that has no value set.

Reply:
But here it's different: the option might exists (and a default value set or not), but it's just one of the option definition components (normalizer, lazy closures, ...) which is not set.

Reply:
then 'NoValueSetException' is good to me ;)

Reply:
Better?